fem/datacollection.hpp

```diff
@@ -23,14 +23,122 @@
 namespace mfem
 {
 
+/// Lightweight adaptor over an std::map from strings to pointer to T
+template<typename T>
+class NamedFieldsMap
```

Comment:
This class was refactored from the existing support in ``DataCollection`` for named GridFunctions and QuadratureFunctions. It is also used in ``SidreDataCollection`` for named Attribute fields (``mfem::Array<int>``).  

It can also be used for simple memory management of ``FiniteElementSpace``s and ``FiniteElementCollection``s in mfem mini-apps / user codes.

Reply:
It will be a shorter implementation if this class derives from `std::map<std::string, T*>` - then most of the methods: `begin()`, `end()`, `find()` will just be inherited, including all other methods of `std::map`.

@kennyweiss, what do you think?

Reply:
I think the general recommendations are not to do that without a compelling reason since the container classes weren't designed for inheritance ( e.g. they lack a virtual destructor).  

See:
* https://stackoverflow.com/questions/4353203/thou-shalt-not-inherit-from-stdvector
* https://stackoverflow.com/questions/10477839/c-inheriting-from-stdmap

A different strategy might be to not wrap these functions, and expose them indirectly to the user through: ``NamedFieldMap::GetMap()``.  This would require some coding changes to users of ``DataCollection``s.

Reply:
OK, let's keep things as they are. Thanks.